src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -822,6 +822,16 @@ private void UpdateCallsiteExecutionState(Object callsiteData, RuntimeCore runti
             return compliantTarget;
         }
 
+        private Boolean IsCompatibleReplicationOption(List<ReplicationInstruction> oldOption, List<ReplicationInstruction> newOption)
+        {
+            if (oldOption.Count > 0 && newOption.Count > 0 && oldOption.Count < newOption.Count)
+            {
```

Comment:
I’m not sure I understand this change. Why are you checking equality only for the first option?

Reply:
For those 2 cases where it was failing, as we are checking for all the replication options(to find a match), the first match is found for "cartesian:indices=0" option. Then it finds a match for "zipped:indices=0,1" and it applies zipped replication option(as this is the final option that is tested). Previously, the first option that found a match was applied directly. 
Another case is, we want to accept this new replication option, if it is of the higher rank than the previous option (new option: "cartesian:indices=0" and previous option: "cartesian:indices=0, cartesian:indices=0". The same option but checks for one extra depth level). Since all options are unique, I was checking the first element and the count for the new option to be higher than the first. Also can use, newOption.Count = oldOption.Count +1.

Reply:
I'm definitely having a hard time understanding this one, is it possible @reddyashish you could draw a diagram or do a longer writeup of the problem here and the approach taken to fix it - I know it's kind of a tall order.

Reply:
so - I'm trying to think of cases where this will fail - 
I think naming this method or adding a summary might help - but my interpretation is that this method is used to find `List<ReplicationInstruction> ` where the first instruction matches the previous one but is of a greater depth?

What I cannot figure out or explain yet is what case this is used to avoid? Can you try to sum it up in the summary of the method or in the git here.